NetworkPkg/NvmeOfDxe/NvmeOfCliInterface.c

```diff
@@ -919,6 +931,36 @@ NvmeOfCliProbeCallback (
       CopyMem (Opts->hostnqn, ProbeconnectData->Hostnqn, sizeof (ProbeconnectData->Hostnqn));
     }
   }
+
+  // Fill socket context
+  Private     = (NVMEOF_DRIVER_DATA*)CallbackCtx;
+  AttemptData = &Private->Attempt->Data;
```

Comment:
Can you assume that Private->Attempt is non-NULL and valid?

Reply:
Yes, Private always contains attempt data for which the Probe callback is invoked. Attempt data is populated to Private before invoking spdk_nvme_probe, while processing a connect command from CLI. Same Private is passed as callback parameter to Probe and Attach callback.